z390test/src/main/groovy/org/z390/test/z390Test.groovy

```diff
@@ -0,0 +1,267 @@
+        }
+        if (System.getenv('Z390_PRINT_OUTPUT')) {
+            printOutput = true
+        }
+    }
+    @BeforeEach
+    void setUp() {
+        this.stdout = ""
+        this.stderr = ""
+        this.fileData = [:]
+    }
+    @AfterEach
+    void cleanUp() {
+        if (this.tempDir)
+            this.tempDir.deleteDir()
+    }
+
+    def createTempFile(String fileName, String fileContents, boolean returnExt=true) {
+        /**
+         * Utility method for creating temp option files used in tests
+         */
+        if (!tempDir) {
+            this.tempDir = File.createTempDir()
+            this.tempDir.deleteOnExit()
+        }
+        var fullFileName = pathJoin(tempDir.absolutePath, fileName)
+        println("Creating temp source: ${fullFileName}")
+        // to allow files to reference themselves, include {{fullFileName}} in contents
+        fullFileName = fullFileName.replace("\\", "\\\\")  // support windows files
+        fileContents = fileContents.replaceAll(/\{\{fullFileName}}/, fullFileName)
+        new File(fullFileName).with {
+            createNewFile()
+            write(fileContents)
+        }
+        String ext = fullFileName.substring(fullFileName.lastIndexOf("."))
+       if (!returnExt || ext.toUpperCase() in ['.MLC', '.CBL'])
+            fullFileName = filenameWithoutExtension(fullFileName)
+        return fullFileName
+    }
+
+    String basePath(String... pathItems) {
+        var fullPathItems = [this.project_root, *pathItems]
+        return new File(fullPathItems.join(File.separator)).getCanonicalPath()
+    }
+
+    String basePathRelative(String... pathItems) {
+        var fullPathItems = [this.project_root, *pathItems]
+        return fullPathItems.join(File.separator)
+    }
+
+    static String filenameWithoutExtension(String filename) {
+        String result = filename[0..<filename.lastIndexOf('.')]
+        println(result)
+        return result
+    }
+
+    def static pathJoin(String... pathItems) {
+        return pathItems.join(File.separator)
+    }
+
+    def getEnvList() {
+        def envList = []
+        this.env.each{item -> envList.add("${item.key}=${item.value}")}
+        return envList
+    }
+
+    def clean(String asmFileExcludingExtension) {
+        /**
+         * clear files for module
+         */
+        for (ext in ["PRN", "ERR", "OBJ", "390", "OBJ", "LOG", "LST", "STA"]) {
+            var filename = asmFileExcludingExtension + '.' + ext
+            var deleteFile = new File(filename)
+            if (deleteFile.exists()) {
+                deleteFile.delete()
+            }
+        }
+    }
+
+    def callZ390(String asmFileExcludingExtension, String command, String... args) {
+        println("Executing ${command}: ${asmFileExcludingExtension}")
+        var cmd = ["java", "-classpath", basePath('jar', 'z390.jar'),
+                   '-Xrs', '-Xms150000K', '-Xmx150000K', command, asmFileExcludingExtension, *args].join(" ")
```

Comment:
All current bash scripts that invoke java use the z390 root directory as the "jar" directory. Look in bash/mz390. You will see the variable jardir set to a value, then the java invocation begins "java -classpath ${jardir}/z390.jar ...". The value of the jardir variable is the full path name to the z390 directory. You can see this by setting "debug-1" on bash/mz390 line 10 and running an assembly. In particular, the z390.jar used is the one in the z390 directory, not the one in the z390/jar directory. If I understand line 110 above, the generated cmd value will use the z390.jar in the z390/jar directory. This is different from normal z390 usage. The two z390.jar files should be the same, but you can't know for sure.

Reply:
OK.. learned something today.. I think in reality they will always be the same because the jar is built in the jar folder and then copied to the root but in the name of consistency, will change to the root as per the standard scripts..